### PR TITLE
Promise lifecycle and memory usage.

### PR DESCRIPTION
**Pull request by aviralg**

### CODE REVIEW

File: compose_testable_vignettes.R [hunk 1]
```diff
@@ -46,9 +46,19 @@ rdt.cmd.head <- function(first, path)
     "include.configuration=TRUE,\n",
     "reload.state=", !first, ",\n",
     "block={\n\n",
+    "loadNamespace <- function(package, lib.loc = NULL,\n",
```
Comment (kondziu) [inline compose_testable_vignettes.R]: I think new functions and stuff should be outside of the Rdt call -- no point introducing extra noise, especially in the case of running small  benchmarks like shootout.

File: README.md [hunk 2]
```diff
@@ -3,3 +3,20 @@ Dynamic tracing support for R
 
 # Build status
 [![Build Status](https://travis-ci.org/PRL-PRG/R-dyntrace.svg?branch=master)](https://travis-ci.org/PRL-PRG/R-dyntrace)
+
+# Commands
```
Comment (kondziu) [inline README.md:7]: These are just notes, right?

Comment (aviralg) [reply README.md:7]: Yup

File: compose_testable_vignettes.R (hunk 1 above)
Comment (aviralg) [reply compose_testable_vignettes.R]: Are you sure the code block will always have access to this definition? Just to be on the safer side, I kept it inside.

Comment (aviralg) [reply compose_testable_vignettes.R]: `gcifno` should be inside the code block. We want to enable it, right before we begin tracing and disable it right after we are done tracing. More appropriate locations are `begin_trace` and `end_trace` hooks respectively.

**Pull request merged**